test/components/tableTests.ts

```diff
@@ -294,77 +118,346 @@ describe("Tables", () => {
+      const bigFixedComponent01 = new Mocks.FixedSizeComponent(BIG_COMPONENT_SIZE, BIG_COMPONENT_SIZE);
+      const bigFixedComponent10 = new Mocks.FixedSizeComponent(BIG_COMPONENT_SIZE, BIG_COMPONENT_SIZE);
+      const smallFixedComponent = new Mocks.FixedSizeComponent(SMALL_COMPONENT_SIZE, SMALL_COMPONENT_SIZE);
+
+      const table = new Plottable.Components.Table([
+        [unfixedComponent, bigFixedComponent01],
+        [bigFixedComponent10, smallFixedComponent]
+      ]);
 
-      assert.isNull(table.componentAt(0, 1), "returns null if an empty cell is queried");
-      assert.isNull(table.componentAt(-1, 0), "returns null if a negative row index is passed in");
-      assert.isNull(table.componentAt(0, -1), "returns null if a negative column index is passed in");
-      assert.isNull(table.componentAt(9001, 0), "returns null if a row index larger than the number of rows is passed in");
-      assert.isNull(table.componentAt(0, 9001), "returns null if a column index larger than the number of columns is passed in");
+      const expectedMinWidth = 2 * BIG_COMPONENT_SIZE;
+      const expectedMinHeight = 2 * BIG_COMPONENT_SIZE;
+
+      const constrainedSpaceRequest = table.requestedSpace(0, 0);
+      TestMethods.verifySpaceRequest(constrainedSpaceRequest, expectedMinWidth, expectedMinHeight,
+        "requests enough space for all fixed-size Components when space is constrained");
+
+      const exactlyEnoughSpaceRequest = table.requestedSpace(expectedMinWidth, expectedMinHeight);
+      TestMethods.verifySpaceRequest(exactlyEnoughSpaceRequest, expectedMinWidth, expectedMinHeight,
+        "requests enough space for all fixed-size Components when given exactly enough space");
+
+      const extraSpaceRequest = table.requestedSpace(2 * expectedMinWidth, 2 * expectedMinHeight);
+      TestMethods.verifySpaceRequest(extraSpaceRequest, expectedMinWidth, expectedMinHeight,
+        "requests enough space for all fixed-size Components when given extra space");
+    });
+  });
+
+  describe("layout of constituent Components", () => {
+    const SVG_WIDTH = 300;
+    const SVG_HEIGHT = 300;
+
+    function verifyOrigins(rows: Plottable.Component[][], rowPadding = 0, columnPadding = 0) {
+      let expectedOrigin = {
+        x: 0,
+        y: 0
+      };
+      rows.forEach((row, r) => {
+        let maxHeight = 0;
+        expectedOrigin.x = 0;
+        row.forEach((component, c) => {
+          TestMethods.assertPointsClose(component.origin(), expectedOrigin, 0, `Component at [${r}, ${c}] has the correct origin`);
+          expectedOrigin.x += component.width() + columnPadding;
+          maxHeight = maxHeight > component.height() ? maxHeight : component.height();
+        });
+        expectedOrigin.y += maxHeight + rowPadding;
+      });
+    }
+
+    it("divides available width evenly between non-fixed-width Components", () => {
+      const svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
```

Comment:
Curious why we need this specific `SVG_WIDTH` and `SVG_HEIGHT`.  Since there is a default, I personally think it would be better to rely on the default behavior unless you absolutely need to take control over the width and height of the svg (constrained cases for example)